ios/Video/RCTVideo.swift

```diff
@@ -458,14 +457,17 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
             }
         #endif
 
-        if source.drm != nil || _localSourceEncryptionKeyScheme != nil {
-            _resouceLoaderDelegate = RCTResourceLoaderDelegate(
+        if source.drm?.type != nil {
```

Comment:
@freeboub source.drm is never `nil` is structure with `nils` even if we don't pass anything are you okay with checking `drm.type` ?

Reply:
I am not sure to understand, you want to change 
    let drm: DRMParams?
by 
    let drm: DRMParams

And check only type ?
if that's the case, I agree !

Something like this patch:
```diff
diff --git a/ios/Video/DataStructures/VideoSource.swift b/ios/Video/DataStructures/VideoSource.swift
index e672929f..45e5b2d3 100644
--- a/ios/Video/DataStructures/VideoSource.swift
+++ b/ios/Video/DataStructures/VideoSource.swift
@@ -10,7 +10,7 @@ struct VideoSource {
     let cropEnd: Int64?
     let customMetadata: CustomMetadata?
     /* DRM */
-    let drm: DRMParams?
+    let drm: DRMParams
     var textTracks: [TextTrack] = []
 
     let json: NSDictionary?
@@ -28,7 +28,7 @@ struct VideoSource {
             self.cropStart = nil
             self.cropEnd = nil
             self.customMetadata = nil
-            self.drm = nil
+            self.drm = DRMParams(nil)
             return
         }
         self.json = json
diff --git a/ios/Video/RCTVideo.swift b/ios/Video/RCTVideo.swift
index 908e3eb8..64130675 100644
--- a/ios/Video/RCTVideo.swift
+++ b/ios/Video/RCTVideo.swift
@@ -421,7 +421,7 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
                 "type": _source?.type ?? NSNull(),
                 "isNetwork": NSNumber(value: _source?.isNetwork ?? false),
             ],
-            "drm": source.drm?.json ?? NSNull(),
+            "drm": source.drm.json ?? NSNull(),
             "target": reactTag as Any,
         ])
 ```

Reply:
I mean:
source.drm is never null it will be always struct with some values (if drm is not setted but user all values will be just null). But now as I think I can aloso check `if source.drm.json != nil` bcs if json is null it means that users notting passed to it